cmd/signozschemamigrator/schema_migrator/column.go

```diff
@@ -0,0 +1,305 @@
+}
+
+func (n NullableColumnType) String() string {
+	return fmt.Sprintf("Nullable(%s)", n.ElementType)
+}
+
+// SimpleAggregateFunction represents a simple aggregate function in a column type.
+// It is used to represent the simple aggregate function in the column type.
+// FunctionName is the name of the simple aggregate function.
+// Arguments are the arguments of the simple aggregate function.
+// Example: SimpleAggregateFunction(sum, Int64), where sum is the function name
+// and Int64 is the argument.
+type SimpleAggregateFunction struct {
+	FunctionName string
+	Arguments    []ColumnType
+}
+
+func (s SimpleAggregateFunction) String() string {
+	arguments := make([]string, len(s.Arguments))
+	for i, arg := range s.Arguments {
+		arguments[i] = arg.String()
+	}
+	return fmt.Sprintf("SimpleAggregateFunction(%s, %s)", s.FunctionName, strings.Join(arguments, ", "))
+}
+
+// AggregateFunction represents an aggregate function in a column type.
+// It is used to represent the aggregate function in the column type.
+// FunctionName is the name of the aggregate function.
+// Arguments are the arguments of the aggregate function.
+// Example: AggregateFunction(sum, Int64), where sum is the function name
+// and Int64 is the argument.
+type AggregateFunction struct {
+	FunctionName string
+	Arguments    []ColumnType
+}
+
+func (a AggregateFunction) String() string {
+	arguments := make([]string, len(a.Arguments))
+	for i, arg := range a.Arguments {
+		arguments[i] = arg.String()
+	}
+	return fmt.Sprintf("AggregateFunction(%s, %s)", a.FunctionName, strings.Join(arguments, ", "))
+}
+
+// Column represents a column in a table.
+// It is used to represent the column in the schema definition.
+// Name (mandatory) is the name of the column.
+// Type (mandatory) is the type of the column.
+// Codec is the codec of the column.
+// Default is the default value/expression of the column. It is
+// user responsibility to ensure that the default value is valid
+// for the column type. The migrator will not validate the default
+// value for you.
+// TTL is the ttl of the column.
+// Settings is the settings of the column.
+// Comment is the comment of the column.
+type Column struct {
+	Name     string
+	Type     ColumnType
+	Codec    string
+	Default  string
+	TTL      string
+	Settings ColumnSettings
+	Comment  string
+}
+
+func (c Column) ToSQL() string {
+	var sql strings.Builder
+	sql.WriteString(c.Name)
+	sql.WriteString(" ")
+	sql.WriteString(c.Type.String())
+	if c.Default != "" {
+		sql.WriteString(" DEFAULT ")
+		sql.WriteString(c.Default)
+	}
+	if c.Codec != "" {
+		sql.WriteString(" CODEC(")
+		sql.WriteString(c.Codec)
+		sql.WriteString(")")
+	}
+	if c.TTL != "" {
+		sql.WriteString(" TTL ")
+		sql.WriteString(c.TTL)
+	}
+	if c.Settings != nil {
+		sql.WriteString(" SETTINGS ")
+		sql.WriteString(c.Settings.String())
+	}
+	if c.Comment != "" {
+		sql.WriteString(" COMMENT ")
+		sql.WriteString(c.Comment)
+	}
+	return sql.String()
```

Comment:
add few tests for these functions ?  or you want to address tests separately.

Reply:
The tests are coming in the upcoming PR where the column definition is used.